### DISCUSSION TITLE
OpenID Connect Custom Claims

### DISCUSSION THREAD

**Discussion by vin047** [zitadel/zitadel#4083] [Q&A] [answered]
Does Zitadel support OpenID Connect (OIDC) custom claims? 

I'd like to be able to set a custom claim for users OIDC token via the API or the console.

**Comment by fforootd** [comment, marked as answer]
Hi @vin047 good question. 

Currently we do not support custom claims. Although that is something that we will work soon.
We will work to extend our [actions ](https://docs.zitadel.com/docs/concepts/features/actions) in combination with the [metadata ](https://docs.zitadel.com/docs/guides/manage/customize/user-metadata) storage we have to allow inserting custom claims.

In the meantime, you can already use our metadata as part of your claim in the tokens.
You can read more about how this works [here](https://docs.zitadel.com/docs/guides/manage/customize/user-metadata),
The metadata allows you to store key value stuff inside a user. So it is flexible in regard of what you want tor store.

To make for more flexibility we will extend where actions (custom js code created by you) can be run. This in such a way that you can change the claims of tokens.

In regard to API access. Except the login every feature ZITADEL has is accessible [through our APIs documented here](https://docs.zitadel.com/docs/apis/introduction).

Hope this was helpful

**Reply by vin047** [reply]
Thanks for the prompt reply. I think I should be able to make things work with the metadata functionality. Cheers!

**Reply by fforootd** [reply]
Ok, just let me know if you have more questions.
